Document new failure message argument position in JUnit 4 migration tips (#2810)
junit-team/junit5: documentation/src/docs/asciidoc/user-guide/migration-from-junit4.adoc

* `@Rule` and `@ClassRule` no longer exist; superseded by `@ExtendWith` and
  `@RegisterExtension`
  - See also <<migrating-from-junit4-rule-support>>.

> * Many assertions and assumptions now accept the message as their last parameter instead of the first one

Reviewer: ```suggestion
* Assertions and assumptions in JUnit Jupiter accept the failure message as their last argument instead of the first one.
```

>   - See also <<migrating-from-junit4-message-parameter, New message parameter position>>.

Reviewer: ```suggestion
  - See <<migrating-from-junit4-failure-message-arguments>> for details.
```

Author: Should I omit the name of the anchor link? 
It would be rendered like this:
See [migrating-from-junit4-failure-message-arguments] for details.

Reviewer: Hmmm.... it should actually use the heading associated with the link if the link is in the same document.

Reviewer: See `(see <<running-tests>> for details` earlier in that document.

----
include::{testDir}/example/IgnoredTestsDemo.java[tags=user_guide]
----

> [[migrating-from-junit4-message-parameter]]
> === New message parameter position

Reviewer: ```suggestion
[[migrating-from-junit4-failure-message-arguments]]
=== Failure Message Arguments
```

> The introduction of the classes `Assumptions` and `Assertions` brought a new parameters' order for existing methods. While on JUnit 4 assertion and assumption methods would accept the message parameter as their first one, JUnit 5 implements them with the message as the last parameter.

Reviewer: ```suggestion
The `Assumptions` and `Assertions` classes in JUnit Jupiter declare arguments in a different order than in JUnit 4. In JUnit 4 assertion and assumption methods accept the failure message as the first argument; whereas, in JUnit Jupiter assertion and assumption methods accept the failure message as the last argument.
```

> For instance, the method `assertEquals` in JUnit 4 is `assertEquals(String message, Object expected, Object actual)` and in JUnit 5 `assertEquals(Object expected, Object actual, String message)`.

Reviewer: ```suggestion
For instance, the method `assertEquals` in JUnit 4 is declared as `assertEquals(String message, Object expected, Object actual)`, but in JUnit Jupiter it is declared as `assertEquals(Object expected, Object actual, String message)`. The rationale for this is that a failure message is _optional_, and optional arguments should be declared after required arguments in a method signature.
```
